### PR TITLE
[16.0][MIG] website_sale_order_type: Migration to version 16.0

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT44380

@stefan-tecnativa @chienandalu please review :)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This one shouldn't be needed if the field is computed writable (which I think it is), but if not, the patch should be to make it on the base module, like we have done with this one: https://github.com/OCA/sale-workflow/commit/788e2bc1d19261c0b7a2d15f10846007230afba2

**Comment by chienandalu** [comment]
> This one shouldn't be needed if the field is computed writable (which I think it is), but if not, the patch should be to make it on the base module

Yes, this can be a good chance to fix that

**Comment by pilarvargas-tecnativa** [comment]
Fix made in https://github.com/OCA/sale-workflow/pull/2639

**Comment by pilarvargas-tecnativa** [comment]
This migration PR has been reopened as the fix made in sale_order_type did not cover the e-commerce flow. Therefore this module is still needed.

@pedrobaeza can you check it too?

I have decided to extend the methods that set the values that are overwritten in the e-commerce flow instead of the method that prepares the dictionary and calls these methods so that in case they are called from anywhere else the intended effect of this module is still in effect.

File: website_sale_order_type/models/website.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class Website(models.Model):
+    _inherit = "website"
+    _description = "Allowing sale_order_type to work with website_sale."
+
+    def sale_get_payment_term(self, partner):
+        payment_term_id = super().sale_get_payment_term(partner)
+        if partner.sale_type.payment_term_id:
+            payment_term_id = partner.sale_type.payment_term_id.id
+        return payment_term_id
```
Comment (pedrobaeza) [inline website_sale_order_type/models/website.py]: And the same for the other:

```suggestion
        if partner.sale_type.payment_term_id:
            return partner.sale_type.payment_term_id.id
        return super().sale_get_payment_term(partner)
```

**Review by pedrobaeza — COMMENTED** [review]
OK for overriding these methods instead.

File: website_sale_order_type/models/website.py (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply website_sale_order_type/models/website.py]: thanks!

**Review by pedrobaeza — APPROVED** [review]

File: website_sale_order_type/models/website.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class Website(models.Model):
+    _inherit = "website"
+    _description = "Allowing sale_order_type to work with website_sale."
```
Comment (chienandalu) [inline website_sale_order_type/models/website.py]: You shouldn't change the model `_description`

**Review by chienandalu — APPROVED** [review]
:+1: 

Just a minor comment:

File: website_sale_order_type/models/website.py (hunk 2 above)
Comment (pedrobaeza) [reply website_sale_order_type/models/website.py]: Good catch! This is inherited by the migration on v15, and provokes a bad problem, renaming the whole model, and sending the button in mails with such text.

**Review by stefan-tecnativa — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
`_description` should be removed.

File: website_sale_order_type/models/website.py (hunk 2 above)
Comment (pilarvargas-tecnativa) [reply website_sale_order_type/models/website.py]: Done, thanks!

**Review by chienandalu — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**